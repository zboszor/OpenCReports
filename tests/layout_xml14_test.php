<?php
/*
 * OpenCReports test
 * Copyright (C) 2019-2024 Zoltán Böszörményi <zboszor@gmail.com>
 * See COPYING.LGPLv3 in the toplevel directory.
 */

$array = [
	[ "id", "name", "property", "age", "adult" ],
	[ "1", "Fred Flintstone", "strong", "31", "yes" ],
	[ "2", "Wilma Flintstone", "charming", "28", "yes" ],
	[ "3", "Pebbles Flintstone", "young", "5e-1", "no" ],
	[ "4", "Betty Rubble", "beautiful", "27", "yes" ],
	[ "5", "Barney Rubble", "small", "29", "yes" ]
];

$coltypes = [
	OpenCReport::RESULT_NUMBER, OpenCReport::RESULT_STRING, OpenCReport::RESULT_STRING, OpenCReport::RESULT_NUMBER, OpenCReport::RESULT_NUMBER
];

$array2 = [
	[ "id", "name", "property", "age", "adult" ],
	[ "1", "Mr. George Slate", "grumpy", "53", "yes" ],
	[ "2", "Joe Rockhead", "friendly", "33", "yes" ],
	[ "3", "Sam Slagheap", "leader", "37", "yes" ],
	[ "4", "The Great Gazoo", "alien", "800", "yes" ],
];

$o = new OpenCReport();
if (!$o->parse_xml("layout_xml14_test.xml")) {
	echo "XML parse error" . PHP_EOL;
	exit(0);
}

$o->set_output_format(OpenCReport::OUTPUT_PDF);

$o->execute();
$o->spool();
