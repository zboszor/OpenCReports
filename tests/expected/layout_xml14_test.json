{
    "part": {
        "pr": {
            "pd": {
                "report": {
                    "reportheader": [
                        {
                            "type": "horizontal_line",
                            "color": "#ff0000",
                            "indent": 0.0,
                            "size": 1.0,
                            "length": 566.47559055118108517
                        },
                        {
                            "type": "line",
                            "elements": [
                                {
                                    "type": "text",
                                    "width": 566.47559055118108517,
                                    "font_point": 14.0,
                                    "font_face": "Courier",
                                    "bold": false,
                                    "italic": false,
                                    "align": "left",
                                    "bgcolor": "#ffffff",
                                    "color": "#000000",
                                    "data": "Flintstones report 1 header"
                                }
                            ]
                        },
                        {
                            "type": "horizontal_line",
                            "color": "#ffffff",
                            "indent": 0.0,
                            "size": 4.0,
                            "length": 566.47559055118108517
                        },
                        {
                            "type": "horizontal_line",
                            "color": "#000000",
                            "indent": 0.0,
                            "size": 2.0,
                            "length": 566.47559055118108517
                        },
                        {
                            "type": "horizontal_line",
                            "color": "#ffffff",
                            "indent": 0.0,
                            "size": 4.0,
                            "length": 566.47559055118108517
                        }
                    ],
                    "fieldheader": [
                        {
                            "type": "horizontal_line",
                            "color": "#000000",
                            "indent": 0.0,
                            "size": 1.0,
                            "length": 566.47559055118108517
                        },
                        {
                            "type": "line",
                            "elements": [
                                {
                                    "type": "text",
                                    "width": 28.80078125,
                                    "font_point": 12.0,
                                    "font_face": "Courier",
                                    "bold": false,
                                    "italic": false,
                                    "align": "left",
                                    "bgcolor": "#e5e5e5",
                                    "color": "#000000",
                                    "data": "ID"
                                },
                                {
                                    "type": "text",
                                    "width": 7.2001953125,
                                    "font_point": 12.0,
                                    "font_face": "Courier",
                                    "bold": false,
                                    "italic": false,
                                    "align": "left",
                                    "bgcolor": "#e5e5e5",
                                    "color": "#000000",
                                    "data": ""
                                },
                                {
                                    "type": "text",
                                    "width": 144.00390625,
                                    "font_point": 12.0,
                                    "font_face": "Courier",
                                    "bold": false,
                                    "italic": false,
                                    "align": "left",
                                    "bgcolor": "#e5e5e5",
                                    "color": "#000000",
                                    "data": "Name"
                                },
                                {
                                    "type": "text",
                                    "width": 7.2001953125,
                                    "font_point": 12.0,
                                    "font_face": "Courier",
                                    "bold": false,
                                    "italic": false,
                                    "align": "left",
                                    "bgcolor": "#e5e5e5",
                                    "color": "#000000",
                                    "data": ""
                                },
                                {
                                    "type": "text",
                                    "width": 72.001953125,
                                    "font_point": 12.0,
                                    "font_face": "Courier",
                                    "bold": false,
                                    "italic": false,
                                    "align": "left",
                                    "bgcolor": "#e5e5e5",
                                    "color": "#000000",
                                    "data": "Property"
                                },
                                {
                                    "type": "text",
                                    "width": 7.2001953125,
                                    "font_point": 12.0,
                                    "font_face": "Courier",
                                    "bold": false,
                                    "italic": false,
                                    "align": "left",
                                    "bgcolor": "#e5e5e5",
                                    "color": "#000000",
                                    "data": ""
                                },
                                {
                                    "type": "text",
                                    "width": 43.201171875,
                                    "font_point": 12.0,
                                    "font_face": "Courier",
                                    "bold": false,
                                    "italic": false,
                                    "align": "left",
                                    "bgcolor": "#e5e5e5",
                                    "color": "#000000",
                                    "data": "Age"
                                },
                                {
                                    "type": "text",
                                    "width": 7.2001953125,
                                    "font_point": 12.0,
                                    "font_face": "Courier",
                                    "bold": false,
                                    "italic": false,
                                    "align": "left",
                                    "bgcolor": "#e5e5e5",
                                    "color": "#000000",
                                    "data": ""
                                },
                                {
                                    "type": "text",
                                    "width": 36.0009765625,
                                    "font_point": 12.0,
                                    "font_face": "Courier",
                                    "bold": false,
                                    "italic": false,
                                    "align": "left",
                                    "bgcolor": "#e5e5e5",
                                    "color": "#000000",
                                    "data": "Adult"
                                }
                            ]
                        },
                        {
                            "type": "horizontal_line",
                            "color": "#000000",
                            "indent": 0.0,
                            "size": 1.0,
                            "length": 566.47559055118108517
                        },
                        {
                            "type": "horizontal_line",
                            "color": "#ffffff",
                            "indent": 0.0,
                            "size": 4.0,
                            "length": 566.47559055118108517
                        }
                    ],
                    "fielddetails": [
                        {
                            "type": "line",
                            "elements": [
                                {
                                    "type": "text",
                                    "width": 28.80078125,
                                    "font_point": 12.0,
                                    "font_face": "Courier",
                                    "bold": false,
                                    "italic": false,
                                    "align": "right",
                                    "bgcolor": "#e5e5e5",
                                    "color": "#000000",
                                    "data": "1"
                                },
                                {
                                    "type": "text",
                                    "width": 7.2001953125,
                                    "font_point": 12.0,
                                    "font_face": "Courier",
                                    "bold": false,
                                    "italic": false,
                                    "align": "left",
                                    "bgcolor": "#e5e5e5",
                                    "color": "#000000",
                                    "data": ""
                                },
                                {
                                    "type": "text",
                                    "width": 144.00390625,
                                    "font_point": 12.0,
                                    "font_face": "Courier",
                                    "bold": false,
                                    "italic": false,
                                    "align": "left",
                                    "bgcolor": "#e5e5e5",
                                    "color": "#000000",
                                    "data": "Mr. George Slate"
                                },
                                {
                                    "type": "text",
                                    "width": 7.2001953125,
                                    "font_point": 12.0,
                                    "font_face": "Courier",
                                    "bold": false,
                                    "italic": false,
                                    "align": "left",
                                    "bgcolor": "#e5e5e5",
                                    "color": "#000000",
                                    "data": ""
                                },
                                {
                                    "type": "text",
                                    "width": 72.001953125,
                                    "font_point": 12.0,
                                    "font_face": "Courier",
                                    "bold": false,
                                    "italic": false,
                                    "align": "left",
                                    "bgcolor": "#e5e5e5",
                                    "color": "#000000",
                                    "data": "grumpy"
                                },
                                {
                                    "type": "text",
                                    "width": 7.2001953125,
                                    "font_point": 12.0,
                                    "font_face": "Courier",
                                    "bold": false,
                                    "italic": false,
                                    "align": "left",
                                    "bgcolor": "#e5e5e5",
                                    "color": "#000000",
                                    "data": ""
                                },
                                {
                                    "type": "text",
                                    "width": 43.201171875,
                                    "font_point": 12.0,
                                    "font_face": "Courier",
                                    "bold": false,
                                    "italic": false,
                                    "align": "right",
                                    "bgcolor": "#e5e5e5",
                                    "color": "#000000",
                                    "data": "53.00"
                                },
                                {
                                    "type": "text",
                                    "width": 7.2001953125,
                                    "font_point": 12.0,
                                    "font_face": "Courier",
                                    "bold": false,
                                    "italic": false,
                                    "align": "left",
                                    "bgcolor": "#e5e5e5",
                                    "color": "#000000",
                                    "data": ""
                                },
                                {
                                    "type": "text",
                                    "width": 36.0009765625,
                                    "font_point": 12.0,
                                    "font_face": "Courier",
                                    "bold": false,
                                    "italic": false,
                                    "align": "center",
                                    "bgcolor": "#e5e5e5",
                                    "color": "#000000",
                                    "data": "yes"
                                }
                            ]
                        }
                    ],
                    "fielddetails": [
                        {
                            "type": "line",
                            "elements": [
                                {
                                    "type": "text",
                                    "width": 28.80078125,
                                    "font_point": 12.0,
                                    "font_face": "Courier",
                                    "bold": false,
                                    "italic": false,
                                    "align": "right",
                                    "bgcolor": "#ffffff",
                                    "color": "#000000",
                                    "data": "2"
                                },
                                {
                                    "type": "text",
                                    "width": 7.2001953125,
                                    "font_point": 12.0,
                                    "font_face": "Courier",
                                    "bold": false,
                                    "italic": false,
                                    "align": "left",
                                    "bgcolor": "#ffffff",
                                    "color": "#000000",
                                    "data": ""
                                },
                                {
                                    "type": "text",
                                    "width": 144.00390625,
                                    "font_point": 12.0,
                                    "font_face": "Courier",
                                    "bold": false,
                                    "italic": false,
                                    "align": "left",
                                    "bgcolor": "#ffffff",
                                    "color": "#000000",
                                    "data": "Joe Rockhead"
                                },
                                {
                                    "type": "text",
                                    "width": 7.2001953125,
                                    "font_point": 12.0,
                                    "font_face": "Courier",
                                    "bold": false,
                                    "italic": false,
                                    "align": "left",
                                    "bgcolor": "#ffffff",
                                    "color": "#000000",
                                    "data": ""
                                },
                                {
                                    "type": "text",
                                    "width": 72.001953125,
                                    "font_point": 12.0,
                                    "font_face": "Courier",
                                    "bold": false,
                                    "italic": false,
                                    "align": "left",
                                    "bgcolor": "#ffffff",
                                    "color": "#000000",
                                    "data": "friendly"
                                },
                                {
                                    "type": "text",
                                    "width": 7.2001953125,
                                    "font_point": 12.0,
                                    "font_face": "Courier",
                                    "bold": false,
                                    "italic": false,
                                    "align": "left",
                                    "bgcolor": "#ffffff",
                                    "color": "#000000",
                                    "data": ""
                                },
                                {
                                    "type": "text",
                                    "width": 43.201171875,
                                    "font_point": 12.0,
                                    "font_face": "Courier",
                                    "bold": false,
                                    "italic": false,
                                    "align": "right",
                                    "bgcolor": "#ffffff",
                                    "color": "#000000",
                                    "data": "33.00"
                                },
                                {
                                    "type": "text",
                                    "width": 7.2001953125,
                                    "font_point": 12.0,
                                    "font_face": "Courier",
                                    "bold": false,
                                    "italic": false,
                                    "align": "left",
                                    "bgcolor": "#ffffff",
                                    "color": "#000000",
                                    "data": ""
                                },
                                {
                                    "type": "text",
                                    "width": 36.0009765625,
                                    "font_point": 12.0,
                                    "font_face": "Courier",
                                    "bold": false,
                                    "italic": false,
                                    "align": "center",
                                    "bgcolor": "#ffffff",
                                    "color": "#000000",
                                    "data": "yes"
                                }
                            ]
                        }
                    ],
                    "fielddetails": [
                        {
                            "type": "line",
                            "elements": [
                                {
                                    "type": "text",
                                    "width": 28.80078125,
                                    "font_point": 12.0,
                                    "font_face": "Courier",
                                    "bold": false,
                                    "italic": false,
                                    "align": "right",
                                    "bgcolor": "#e5e5e5",
                                    "color": "#000000",
                                    "data": "3"
                                },
                                {
                                    "type": "text",
                                    "width": 7.2001953125,
                                    "font_point": 12.0,
                                    "font_face": "Courier",
                                    "bold": false,
                                    "italic": false,
                                    "align": "left",
                                    "bgcolor": "#e5e5e5",
                                    "color": "#000000",
                                    "data": ""
                                },
                                {
                                    "type": "text",
                                    "width": 144.00390625,
                                    "font_point": 12.0,
                                    "font_face": "Courier",
                                    "bold": false,
                                    "italic": false,
                                    "align": "left",
                                    "bgcolor": "#e5e5e5",
                                    "color": "#000000",
                                    "data": "Sam Slagheap"
                                },
                                {
                                    "type": "text",
                                    "width": 7.2001953125,
                                    "font_point": 12.0,
                                    "font_face": "Courier",
                                    "bold": false,
                                    "italic": false,
                                    "align": "left",
                                    "bgcolor": "#e5e5e5",
                                    "color": "#000000",
                                    "data": ""
                                },
                                {
                                    "type": "text",
                                    "width": 72.001953125,
                                    "font_point": 12.0,
                                    "font_face": "Courier",
                                    "bold": false,
                                    "italic": false,
                                    "align": "left",
                                    "bgcolor": "#e5e5e5",
                                    "color": "#000000",
                                    "data": "leader"
                                },
                                {
                                    "type": "text",
                                    "width": 7.2001953125,
                                    "font_point": 12.0,
                                    "font_face": "Courier",
                                    "bold": false,
                                    "italic": false,
                                    "align": "left",
                                    "bgcolor": "#e5e5e5",
                                    "color": "#000000",
                                    "data": ""
                                },
                                {
                                    "type": "text",
                                    "width": 43.201171875,
                                    "font_point": 12.0,
                                    "font_face": "Courier",
                                    "bold": false,
                                    "italic": false,
                                    "align": "right",
                                    "bgcolor": "#e5e5e5",
                                    "color": "#000000",
                                    "data": "37.00"
                                },
                                {
                                    "type": "text",
                                    "width": 7.2001953125,
                                    "font_point": 12.0,
                                    "font_face": "Courier",
                                    "bold": false,
                                    "italic": false,
                                    "align": "left",
                                    "bgcolor": "#e5e5e5",
                                    "color": "#000000",
                                    "data": ""
                                },
                                {
                                    "type": "text",
                                    "width": 36.0009765625,
                                    "font_point": 12.0,
                                    "font_face": "Courier",
                                    "bold": false,
                                    "italic": false,
                                    "align": "center",
                                    "bgcolor": "#e5e5e5",
                                    "color": "#000000",
                                    "data": "yes"
                                }
                            ]
                        }
                    ],
                    "fielddetails": [
                        {
                            "type": "line",
                            "elements": [
                                {
                                    "type": "text",
                                    "width": 28.80078125,
                                    "font_point": 12.0,
                                    "font_face": "Courier",
                                    "bold": false,
                                    "italic": false,
                                    "align": "right",
                                    "bgcolor": "#ffffff",
                                    "color": "#000000",
                                    "data": "4"
                                },
                                {
                                    "type": "text",
                                    "width": 7.2001953125,
                                    "font_point": 12.0,
                                    "font_face": "Courier",
                                    "bold": false,
                                    "italic": false,
                                    "align": "left",
                                    "bgcolor": "#ffffff",
                                    "color": "#000000",
                                    "data": ""
                                },
                                {
                                    "type": "text",
                                    "width": 144.00390625,
                                    "font_point": 12.0,
                                    "font_face": "Courier",
                                    "bold": false,
                                    "italic": false,
                                    "align": "left",
                                    "bgcolor": "#ffffff",
                                    "color": "#000000",
                                    "data": "The Great Gazoo"
                                },
                                {
                                    "type": "text",
                                    "width": 7.2001953125,
                                    "font_point": 12.0,
                                    "font_face": "Courier",
                                    "bold": false,
                                    "italic": false,
                                    "align": "left",
                                    "bgcolor": "#ffffff",
                                    "color": "#000000",
                                    "data": ""
                                },
                                {
                                    "type": "text",
                                    "width": 72.001953125,
                                    "font_point": 12.0,
                                    "font_face": "Courier",
                                    "bold": false,
                                    "italic": false,
                                    "align": "left",
                                    "bgcolor": "#ffffff",
                                    "color": "#000000",
                                    "data": "alien"
                                },
                                {
                                    "type": "text",
                                    "width": 7.2001953125,
                                    "font_point": 12.0,
                                    "font_face": "Courier",
                                    "bold": false,
                                    "italic": false,
                                    "align": "left",
                                    "bgcolor": "#ffffff",
                                    "color": "#000000",
                                    "data": ""
                                },
                                {
                                    "type": "text",
                                    "width": 43.201171875,
                                    "font_point": 12.0,
                                    "font_face": "Courier",
                                    "bold": false,
                                    "italic": false,
                                    "align": "right",
                                    "bgcolor": "#ffffff",
                                    "color": "#000000",
                                    "data": "800.00"
                                },
                                {
                                    "type": "text",
                                    "width": 7.2001953125,
                                    "font_point": 12.0,
                                    "font_face": "Courier",
                                    "bold": false,
                                    "italic": false,
                                    "align": "left",
                                    "bgcolor": "#ffffff",
                                    "color": "#000000",
                                    "data": ""
                                },
                                {
                                    "type": "text",
                                    "width": 36.0009765625,
                                    "font_point": 12.0,
                                    "font_face": "Courier",
                                    "bold": false,
                                    "italic": false,
                                    "align": "center",
                                    "bgcolor": "#ffffff",
                                    "color": "#000000",
                                    "data": "yes"
                                }
                            ]
                        }
                    ],
                    "reportfooter": [
                        {
                            "type": "horizontal_line",
                            "color": "#ff0000",
                            "indent": 0.0,
                            "size": 1.0,
                            "length": 566.47559055118108517
                        },
                        {
                            "type": "line",
                            "elements": [
                                {
                                    "type": "text",
                                    "width": 566.47559055118108517,
                                    "font_point": 14.0,
                                    "font_face": "Courier",
                                    "bold": false,
                                    "italic": false,
                                    "align": "left",
                                    "bgcolor": "#ffffff",
                                    "color": "#000000",
                                    "data": "Flintstones report 1 footer"
                                }
                            ]
                        },
                        {
                            "type": "horizontal_line",
                            "color": "#ffffff",
                            "indent": 0.0,
                            "size": 4.0,
                            "length": 566.47559055118108517
                        },
                        {
                            "type": "horizontal_line",
                            "color": "#000000",
                            "indent": 0.0,
                            "size": 2.0,
                            "length": 566.47559055118108517
                        },
                        {
                            "type": "horizontal_line",
                            "color": "#ffffff",
                            "indent": 0.0,
                            "size": 4.0,
                            "length": 566.47559055118108517
                        }
                    ]
                }
            }
        }
    }
}
